File: dishmanagementservice/internal/dto/dish_dto.go
package dto

import "time"

type Address struct {
	Street		string		`json:"street" binding:"required"`
	City		string		`json:"city" binding:"required"`
	State		string		`json:"state" binding:"required"`
	PostalCode	string		`json:"postalCode" binding:"required"`
	Country		string		`json:"country" binding:"required"`
}

type GeoPoint struct {
	Latitude	float64		`json:"latitude"`
	Longitude	float64		`json:"longitude"`
}

type CreateDishRequest struct {
	DishName			string		`json:"dishName" binding:"required"`
	ChefID				string		`json:"chefId" binding:"required"`
	Description			string		`json:"description" binding:"required"`
	Price				float64		`json:"price" binding:"required,gt=0"`
	AvailablePortions	int			`json:"availablePortions" binding:"required,gt=0"`
	MealCourse			string		`json:"mealCourse"`
	DietaryCategory		string		`json:"dietaryCategory"`
	AvailableUntil		time.Time	`json:"availableUntil"`
	Address				Address		`json:"address" binding:"required"`
}


type UpdateDishRequest struct {
	DishID			  string	`json:"dishId" binding:"required"`
	DishName          string    `json:"dishName" binding:"required"`
	ChefID            string    `json:"chefId" binding:"required"`
	Description       string    `json:"description"`
	Price             float64   `json:"price" binding:"gt=0"`
	AvailablePortions int       `json:"availablePortions" binding:"gt=0"`
	MealCourse        string    `json:"mealCourse"`
	DietaryCategory   string    `json:"dietaryCategory"`
	AvailableUntil    time.Time `json:"availableUntil"`
	Address           Address   `json:"address"`
}

type DishResponse struct {
	DishID				string		`json:"dishId"`
	DishName			string		`json:"dishName"`
	ChefID				string		`json:"chefId" binding:"required"`
	Description			string		`json:"description"`
	Price				float64		`json:"price" binding:"gt=0"`
	AvailablePortions	int			`json:"availablePortions" binding:"gt=0"`
	MealCourse			string		`json:"mealCourse"`
	DietaryCategory		string		`json:"dietaryCategory"`
	AvailableUntil		time.Time	`json:"availableUntil"`
	Address				Address		`json:"address"`
	GeoPoint			GeoPoint	`json:"geoPoint"`
	CreatedAt			time.Time	`json:"createdAt"`
	UpdatedAt			time.Time	`json:"updatedAt"`
}

File: dishmanagementservice/src/main/java/edu/sjsu/kairos/dishmanagementservice/dto/DishDTO.java
package edu.sjsu.kairos.dishmanagementservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class DishDTO {

    @JsonProperty("dish_id")
    private UUID dishId;

    @JsonProperty("dish_name")
    private String dishName;

    @JsonProperty("chef_id")
    private String chefId;

    @JsonProperty("price")
    private double price;

    @JsonProperty("available_portions")
    private int availablePortions;

    @JsonProperty("description")
    private String description;

    @JsonProperty("meal_course")
    private String mealCourse;

    @JsonProperty("dietary_category")
    private String dietaryCategory;

    @JsonProperty("ingredients")
    private String ingredients;

    @JsonProperty("available_until")
    private Instant availableUntil;

    @JsonProperty("address")
    private AddressDTO address;

    @JsonProperty("location")
    private LocationDTO location;

    @JsonProperty("images")
    private List<ImageDTO> images;

    @JsonProperty("created_at")
    private Instant createdAt;

    @JsonProperty("updated_at")
    private Instant updatedAt;
}
